File: abc160/d/d.js
function main(input) {
  const inp = input.split(' ');
  const n = Number(inp[0]);
  const x = Number(inp[1]);
  const y = Number(inp[2]);

  function dist(a, b) {
    return Math.min(b-a, Math.abs(x-a)+Math.abs(y-b)+1);
  }

  for (var i=1; i<n; i++) {
    var cnt = 0;

    for (var a=1; a<=n-i; a++) {
      var s = a+i;
      if (s <= n && dist(a,s) === i)
        cnt++; 

      var rmn = i - 1 - Math.abs(x-a);

      if (rmn > 0) {
        var prev = y - rmn;
        if (prev !== s && dist(a, prev) === i)
          cnt++; 

        var next = y + rmn;
        if (next !== s && next <= n && dist(a, next) === i)
          cnt++; 
      } else if(rmn === 0 && y !== s && dist(a, y) === i)
        cnt++; 
    }

    console.log(cnt);
  }
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: abc160/d/d.rb
n, $x, $y = gets.chomp.split(' ').map(&:to_i)

def dist(a, b)
  a - b < ($x - a).abs + ($y - b).abs + 1 ? a - b : ($x - a).abs + ($y - b).abs + 1
end

(1..n-1).each do |i|
  count = 0
  (1..n-1).each do |j|
    b = i + j
    count += 1 if dist(j, b) == i

    rmn = i - ($x - j).abs - 1

    if rmn > 0
      back = $y - rmn
      count += 1 if back != b && dist(j, back)

      forward = $y + rmn
      count += 1 if forward != b && dist(j, forward)
    elsif rmn == 0 && $y != b && dist(j, $y) == i
      count += 1
    end
  end

  puts count
end